packages/acceptance-tests/pkg-tests-specs/sources/legacy.test.ts

```diff
@@ -88,4 +88,35 @@ describe(`Legacy tests`, () => {
       },
     ),
   );
+
+  test(
+    `it should import unconventional tarball entries`,
+    makeTemporaryEnv(
+      {
+        dependencies: {[`unconventional-tarball`]: `^1.0.0`},
+      },
+      async ({path, run, source}) => {
+        await xfs.writeFilePromise(
+          `${path}/yarn.lock` as PortablePath,
+          `
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+"unconventional-tarball@^1.0.0":
+  version "1.0.0"
+  resolved "https://npm.pkg.github.com/download/@private/unconventional-tarball/1.0.0/ccbefc8f730508fc32a91ebb41b7d71417b38f3079c55b7d8039fb8cac9a4fae"
```

Comment:
Can you replace that by a call to `getPackageHttpArchivePath` so that it doesn't rely on the network? You should be able to append a query string to make it fail the "conventional url" check.

Reply:
It doesn't make a network call to the GitHub registry, the `LegacyMigrationResolver` doesn't preserve that information.